packages/server/src/next-utils.ts

```diff
@@ -253,30 +253,25 @@ export function startCustomServer(
         })
     }
 
-    if (watch) {
-      // Handle build & Starting server
-      const esbuildOptions = getEsbuildOptions()
-      esbuildOptions.watch = {
-        onRebuild(error) {
-          if (error) {
-            log.error("Failed to re-build custom server")
-          } else {
-            newline()
-            log.progress("Custom server changed - restarting...")
-            newline()
-            //@ts-ignore -- incorrect TS type from node
-            process.exitCode = RESTART_CODE
-            process.kill("SIGABRT")
-          }
-        },
-      }
-      // eslint-disable-next-line @typescript-eslint/no-floating-promises
-      esbuild.build(esbuildOptions).then(() => {
-        spawnServer()
-      })
-    } else {
-      // No build required, Start server
+    // Handle build & Starting server
+    const esbuildOptions = getEsbuildOptions()
+    esbuildOptions.watch = watch ? {
+      onRebuild(error) {
+        if (error) {
+          log.error("Failed to re-build custom server")
+        } else {
+          newline()
+          log.progress("Custom server changed - restarting...")
+          newline()
+          //@ts-ignore -- incorrect TS type from node
+          process.exitCode = RESTART_CODE
+          process.kill("SIGABRT")
+        }
+      },
+    } : undefined
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    esbuild.build(esbuildOptions).then(() => {
       spawnServer()
-    }
+    })
```

Comment:
Thanks @andreasasprou!

This change has a slight regression.
- Before: `blitz start` will not build custom server (it was built with `blitz build`)
- After: `blitz start` will build custom server, requiring dev deps in production environment.

So we need to skip esbuild if `config.env === 'prod'` in this function.

Reply:
I'd added a small change to mitigate this in the most recent commit in this PR